warehouse/query-core/src/test/java/datawave/query/tables/keyword/KeywordQueryTransformerTest.java

```diff
@@ -0,0 +1,110 @@
+package datawave.query.tables.keyword;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Test;
+
+import datawave.microservice.query.Query;
+import datawave.microservice.query.QueryImpl;
+import datawave.query.transformer.KeywordQueryTransformer;
+import datawave.webservice.query.result.event.Metadata;
+
+public class KeywordQueryTransformerTest {
+
+    String[] input = {"DOCUMENT:shard1/datatype/uid", "DOCUMENT:shard2/datatype/uid!PAGE_ID:12345",
+            "DOCUMENT:shard3/datatype/uid!PAGE_ID:12345%LANGUAGE:ENGLISH", "DOCUMENT:shard4/datatype/uid%LANGUAGE:ENGLISH",
+
+    };
+
+    String[][] expected = {{"DOCUMENT", "shard1", "datatype", "uid", "", ""}, {"DOCUMENT", "shard2", "datatype", "uid", "PAGE_ID:12345", ""},
+            {"DOCUMENT", "shard3", "datatype", "uid", "PAGE_ID:12345", "ENGLISH"}, {"DOCUMENT", "shard4", "datatype", "uid", "", "ENGLISH"},};
+
+    String[] malformedInput = {"DOCUMENT:shard1/datatype", "shard2/datatype", "DOCUMENT:shard3/data!type/uid!PAGE_ID:12345%LANGUAGE:ENGLISH",
+            "shard5/datatype/uid!PAGE_ID:54321", "shard6/datatype/uid%LANGUAGE:ENGLISH", "shard7/datatype/uid!PAGE_ID:54321%LANGUAGE:ENGLISH"};
```

Comment:
Would be nice to see these on separate lines via disabling the formatter

Reply:
This is fixed in the feature/keyword-extraction-chain branch. Thanks.